imap_processing/swe/l1a/swe_l1a.py

```diff
@@ -13,15 +21,51 @@ def swe_l1a(packet_file: str):
 
     Parameters
     ----------
-    packet_file : str
-        The path and filename to the L0 file to read
+    packets: list
+        Decom data list that contains all appIds
+
+    Returns
+    -------
+    str
+        Path name of where CDF file was created.
+        This is used to upload file from local to s3.
+        TODO: test this later.
     """
-    decom_data = decom_swe.decom_packets(packet_file)
-    logging.info(f"Unpacking data from {packet_file}")
-
-    # If appId is science, then the file should contain all data of science appId
-    if decom_data[0].header["PKT_APID"].raw_value == 1344:
-        logging.info("Processing science data")
-        return swe_science(decom_data=decom_data)
-    else:
-        return decom_data
+    # group data by appId
+    grouped_data = group_by_apid(packets)
+
+    for apid in grouped_data.keys():
+        # If appId is science, then the file should contain all data of science appId
+        if apid == SWEAPID.SWE_SCIENCE.value:
+            # sort data by acquisition time
+            sorted_packets = sort_by_time(grouped_data[apid], "ACQ_START_COARSE")
+            logging.debug(
+                "Processing science data for [%s] packets", len(sorted_packets)
+            )
+            data = swe_science(decom_data=sorted_packets)
+        else:
+            # If it's not science, we unpack, organize and save it as a dataset.
+            sorted_packets = sort_by_time(grouped_data[apid], "SHCOARSE")
+            data = create_dataset(packets=sorted_packets)
+
+        current_dir = Path(__file__).parent
+        # write data to CDF
+        if apid == SWEAPID.SWE_APP_HK.value:
+            return write_to_cdf(
+                data,
+                "swe",
+                "l1a",
+                version=__version__,
+                mode=f"{data['APP_MODE'].data[0]}",
+                description=get_descriptor(apid),
+                directory=current_dir,
+            )
+        else:
+            return write_to_cdf(
+                data,
+                "swe",
+                "l1a",
+                version=__version__,
+                description=get_descriptor(apid),
+                directory=current_dir,
+            )
```

Comment:
Put the only difference into an if statement and return after to shorten the implementation?

```suggestion
      return write_to_cdf(
          data,
          "swe",
          "l1a",
          version=__version__,
          mode=f"{data['APP_MODE'].data[0]}" if apid == SWEAPID.SWE_APP_HK else "",
          description=get_descriptor(apid),
          directory=current_dir,
      )
```